### PR TITLE
Remove ToString from RuntimeAssembly.GetManifestResourceStream

### PR DESCRIPTION
**Pull request by stephentoub**
A small allocation we can avoid with the span-based string.Concat.

### CODE REVIEW

**Comment by filipnavara** [comment]
There's one more instance of the same code in `ManifestBasedResourceGroveler`. I have a refactoring PR in queue to merge them (#21979).

File: src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs [hunk 1]
```diff
@@ -207,8 +207,11 @@ public override Stream GetManifestResourceStream(Type type, string name)
                 throw new ArgumentNullException(nameof(type));
 
             string nameSpace = type?.Namespace;
-            string delimiter = (nameSpace != null && name != null) ? Type.Delimiter.ToString() : null;
-            string resourceName = string.Concat(nameSpace, delimiter, name);
+
+            char c = Type.Delimiter;
```
Comment (danmosemsft) [inline src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs]: If you feel like it you could change `nameSpace` to `namespace` ...

Comment (stephentoub) [reply src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs]: Not without putting an @ in front of it, as it's a reserved keyword.

File: src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs [hunk 2]
```diff
@@ -207,8 +207,11 @@ public override Stream GetManifestResourceStream(Type type, string name)
...
+            string resourceName = nameSpace != null && name != null ?
+                string.Concat(nameSpace, new Span<char>(ref c, 1), name) :
```
[the remark this answers is not in the document]
Comment (stephentoub) [reply src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs]: Yup.  I'm pretty sure I've made this mistake in other places, too; I'll do a pass to clean them up.

**Comment by stephentoub** [comment]
> There's one more instance of the same code in ManifestBasedResourceGroveler. I have a refactoring PR in queue to merge them (#21979).

Thanks, @filipnavara.  If I merge this, will that mess up your work?

**Comment by filipnavara** [comment]
>  If I merge this, will that mess up your work?

It's fine, I will rebase the other PR.

**Comment by stephentoub** [comment]
@dotnet-bot test Windows_NT x64 Checked CoreFX Tests please

**Comment by stephentoub** [comment]
> It's fine, I will rebase the other PR.

Ok, thanks!

**Pull request merged**